compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -427,6 +433,95 @@ WholeOutput RecordMinMax::importH5Data(const std::string &input_data_path)
   }
 }
 
+std::unique_ptr<DataSetIterator> RecordMinMax::createIterator()
+{
+  assert(_data_set_format != DataSetFormat::UNKNOWN); // FIX_CALLER_UNLESS
+
+  std::unique_ptr<DataSetIterator> iterator;
+  switch (_data_set_format)
+  {
+    case DataSetFormat::H5:
+      assert(not _input_data_path.empty()); // FIX_CALLER_UNLESS
+      iterator = std::make_unique<HDF5Iterator>(_input_data_path, _module.get());
+      break;
+    case DataSetFormat::RANDOM:
+      iterator = std::make_unique<RandomIterator>(_module.get());
+      break;
+    case DataSetFormat::DIRECTORY:
+      iterator = std::make_unique<DirectoryIterator>(_input_data_path, _module.get());
+      break;
+    case DataSetFormat::LIST_FILE:
+      iterator = std::make_unique<ListFileIterator>(_input_data_path, _module.get());
+      break;
+    default:
+      throw std::runtime_error("Unsupported dataset format");
+  }
+
+  assert(iterator.get() != nullptr); // FIX_ME_UNLESS
+
+  return iterator;
+}
+
+void RecordMinMax::profileData()
```

Comment:
Before this PR, there were four profile functions. This PR enables to use this unified profile function.